### PR TITLE
BA-1904 Display Chat Header

### PR DESCRIPTION
**Pull request by tsl-ps2**
This implements a header for the current chat. In addition to the title described below, it also includes a back button on small screens. 

**Description**

As a user, on the BaseApp Messages, I would like to see a header at the top of the chat room, In order to I can quickly view who am I talking to.

**Acceptance Criteria** 

- Given I am messaging a single person then the chat room header should display the name of the profile
- Given I am messaging a group of people then the chat room header should display the [chat room name/title] in the BE.
- Given the chat room has an associated profile picture, when I view the header, then I should see the profile picture displayed next to the chat room name.
- If the profile does not have a picture, then display a placeholder image for profile.

**Guardrails**

The 3 dots menu in the header will not be worked on right now. 

## Release Notes

- **New Features**
  - Introduced `ChatRoomHeader` component for improved chat room header display.
  - Added `ChatHeaderContainer` styled component for enhanced layout.
  - Implemented `ChatRoomHeaderFragment` for efficient data retrieval in chat room headers.
  - Added utility functions for managing chat room data, including participant count and group chat determination.
  - Introduced `getSubtitle` function for generating subtitles based on participant count.
  - Enhanced `ChatRoom` layout by integrating `ChatRoomHeader`.
  - Updated changelog to reflect new chat header features in version 0.0.33.

- **Bug Fixes**
  - Streamlined logic in `ChatRoomItem` for title and avatar display.

- **Documentation**
  - Updated exports in the messaging module for better query management.

### CODE REVIEW

File: packages/components/modules/messages/hooks/useNameAndAvatar.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { useCurrentProfile } from '@baseapp-frontend/authentication'
+
+import { useFragment } from 'react-relay'
+
+import { ChatRoomHeaderFragment$key } from '../../../__generated__/ChatRoomHeaderFragment.graphql'
+import { ChatRoomHeaderFragment } from '../graphql/queries/ChatRoomHeaderFragment'
+import { getParticipantCount, isGroupChat } from './utils'
+
+const useNameAndAvatar = (roomHeaderRef: ChatRoomHeaderFragment$key) => {
```
Comment (tsl-ps2) [inline packages/components/modules/messages/hooks/useNameAndAvatar.ts]: This is written in a reusable way and should also be used to obtain the chat name and avatar in the chat list. I will update this when the corresponding PRs are merged.

File: packages/components/modules/messages/hooks/utils.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { ChatRoomHeaderFragment$data } from '../../../__generated__/ChatRoomHeaderFragment.graphql'
+
+export const isGroupChat = (chatRoom: ChatRoomHeaderFragment$data) => chatRoom.title !== null
```
Comment (tsl-ps2) [inline packages/components/modules/messages/hooks/utils.ts]: This should be updated when a `isGroupChat` field is added on the backend.

File: packages/components/modules/messages/hooks/utils.ts
Comment (anicioalexandre) [inline packages/components/modules/messages/hooks/utils.ts:1]: `packages/components/modules/messages/utils.ts` would be a better place for this file

File: packages/components/modules/messages/ChatRoom/ChatRoomHeader/index.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+import { ChatRoomHeaderFragment$data } from '../../../../__generated__/ChatRoomHeaderFragment.graphql'
+import { useChatRoom } from '../../context'
+import { ChatRoomHeaderFragment } from '../../graphql/queries/ChatRoomHeaderFragment'
+import useNameAndAvatar from '../../hooks/useNameAndAvatar'
+import { getParticipantCount, isGroupChat } from '../../hooks/utils'
+import { ChatHeaderContainer } from './styled'
+import { ChatRoomHeaderProps } from './types'
+
+const getSubtitle = (roomHeader: ChatRoomHeaderFragment$data) => {
```
Comment (anicioalexandre) [inline packages/components/modules/messages/ChatRoom/ChatRoomHeader/index.tsx]: we normally place this util function under `ChatRoomHeader/utils.ts`

File: packages/components/modules/messages/ChatRoom/ChatRoomHeader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  return null
+}
+
+const ChatRoomHeader: FC<ChatRoomHeaderProps> = ({ roomHeaderRef }) => {
+  const roomHeader = useFragment(ChatRoomHeaderFragment, roomHeaderRef)
+
+  const isUpToMd = useResponsive('up', 'md')
+  const { resetChatRoom } = useChatRoom()
+
+  const { title, avatar } = useNameAndAvatar(roomHeaderRef)
```
Comment (anicioalexandre) [inline packages/components/modules/messages/ChatRoom/ChatRoomHeader/index.tsx]: I've noticed that `useFragment` is being called twice for the same fragment in both the `ChatRoomHeader` component and the `useNameAndAvatar` hook. This results in the same fragment being read twice within the same render tree, introducing unnecessary complexity.

Passing the `roomHeader` object (resolved data from `useFragment`) directly to useNameAndAvatar would probably be the best choice here

File: packages/components/modules/messages/hooks/utils.ts (hunk 2 above)
Comment (anicioalexandre) [reply packages/components/modules/messages/hooks/utils.ts]: can we add a TODO so we don't lose track of it

File: packages/components/modules/messages/ChatRoom/ChatRoomHeader/index.tsx (hunk 3 above)
Comment (tsl-ps2) [reply packages/components/modules/messages/ChatRoom/ChatRoomHeader/index.tsx:21]: @anicioalexandre Then the linter complains that `getSubtitle` is used before it is defined. Should I still move it below and suppress this error or rather keep it where it is?

**Review by anicioalexandre — APPROVED** [review]
just missing versioning

**Review by pt-tsl — APPROVED** [review]

**Pull request merged**